go/internal/cbor/addinfo.go

```diff
@@ -0,0 +1,81 @@
+package cbor
+
+import "fmt"
+
+// 5 bits of the first byte on CBOR contain the additional information which can be either the
+// actual value or a value telling us how to get the value.
+
+// The meaning of additional information depends on the major type. For example, in major type 0,
+// the argument is the value of the data item itself (and in major type 1, the value of the data
+// item is computed from the argument); in major type 2 and 3, it gives the length of the string
+// data in bytes that follow; and in major types 4 and 5, it is used to determine the number of data
+// items enclosed.
+
+// https://www.rfc-editor.org/rfc/rfc8949.html#name-specification-of-the-cbor-e
+
+type AdditionalInfo int
+
+const (
+	// The integers match with the length in bytes apart from reserved and indefinite.
```

Comment:
I think this is confusing. I'd use simple sequential enum values (using `iota`), and use a `switch` statement in `getAdditionalInfoLength()`.